src/item_location.cpp

```diff
@@ -877,13 +900,40 @@ int item_location::max_charges_by_parent_recursive( const item &it ) const
         return item::INFINITE_CHARGES;
     }
 
-    item_location parent = parent_item();
-    item_pocket *pocket = parent_pocket();
+    float weight_multiplier = 1.0f;
+    float volume_multiplier = 1.0f;
+    units::mass max_weight = 1000_kilogram;
+    units::volume max_volume = 1000_liter;
+    item_location current_location = *this;
+    //item_location class cannot return current pocket so use first pocket for innermost container
+    //Only used for weight and volume multipliers
+    const item_pocket *current_pocket = get_item()->get_all_standard_pockets().front();
+
+    //Repeat until top-most container reached
+    while( current_location.has_parent() ) {
+        //Multiply weight and volume multipliers for each container
+        weight_multiplier = weight_multiplier * current_pocket->data->weight_multiplier;
+        volume_multiplier = volume_multiplier * current_pocket->data->volume_multiplier;
+        //Inserting into rigid pockets will not affect parent volume so stop keeping track
+        if( current_pocket->rigid() ) {
+            volume_multiplier = 0.0f;
+        };
+
+        //Calculate effective remaining weight and volume for current parent
+        units::mass temp_weight = weight_capacity() / weight_multiplier;
+        units::volume temp_volume = volume_capacity() / volume_multiplier;
```

Comment:
potentially you divide by 0 due to line 919 and there's items with weight multiplier set to 0 (ebook pockets, magiclysm various bags of holdings), float division won't abort (usually) but you'll get a nan or infinity

Reply:
Thanks, can't believe I missed this